syntax-test_dev.py

```diff
@@ -0,0 +1,271 @@
+                    break
+            else:
+                scopes.append(scope)
+            length += 1
+            if test_at_start_of_comment:
+                break
+
+        # find the unique scopes at each existing assertion position
+        if lines is not None and not test_at_start_of_comment:
+            for pos in range(line.begin() + lines[0][1][0], line.begin() + lines[0][1][1]):
+                scope = view.scope_name(pos)
+                if scope not in scopes:
+                    scopes.append(scope)
+
+        # find the shared scopes
+        # TODO: more clever matching of partial scopes
+        # i.e. meta.function.python, meta.function.parameters.python == meta.function
+        shared_scopes = []
+        for check_scope in scopes[0].split():
+            if all(sublime.score_selector(scope, check_scope) > 0 for scope in scopes):
+                shared_scopes.append(check_scope)
+
+        # skip the first scope if it is the base scope
+        # (crude check by scanning the first scope at the beginning of the file)
+        start_index = 0
+        if shared_scopes[0] == view.scope_name(0).split()[0]:
+            start_index = 1
+
+        scope = ' '.join(shared_scopes[start_index:])
+
+        # delete the existing selection
+        if not view.sel()[0].empty():
+            view.erase(edit, view.sel()[0])
+
+        view.insert(edit, insert_at, (char * length) + ' ' + scope + end_token)
+
+        # move the selection to cover the added scope name,
+        # so that the user can easily insert another ^ to extend the test
+        view.sel().clear()
+        view.sel().add(sublime.Region(
+            insert_at + length,
+            insert_at + length + len(' ' + scope + end_token)
+        ))
+
+
+class HighlightTestViewEventListener(sublime_plugin.ViewEventListener):
+    def on_selection_modified_async(self):
+        """When the selection changes, (re)move the highlight that shows where the current line's
+        test assertions relate to."""
+
+        cursor = self.view.sel()[0]
+        highlight_only_cursor = False
+        if cursor.empty():
+            cursor = sublime.Region(cursor.begin(), cursor.end() + 1)
+        else:
+            highlight_only_cursor = re.match(r'^\^+$', self.view.substr(cursor)) is not None
+
+        lines, line = get_details_of_line_being_tested(self.view)
+
+        if lines is None or not lines[0][1] or not lines[0][0]:
+            self.view.erase_regions('current_syntax_test')
+            return
+
+        col_start, col_end = lines[0][1]
+        if highlight_only_cursor:
+            col_start = self.view.rowcol(cursor.begin())[1]
+            col_end = self.view.rowcol(cursor.end())[1]
+        elif col_end == col_start:
+            col_end += 1
+
+        region = sublime.Region(line.begin() + col_start, line.begin() + col_end)
+        self.view.add_regions('current_syntax_test', [region], 'text', '', sublime.DRAW_NO_FILL)
```

Comment:
Imo this should be configurable. Both the style and the scope name.
I already have highlighting from https://packagecontrol.io/packages/SublimeLinter-contrib-sublime-syntax on test lines, so they *might* conflict (although SL's coloring is almost always red).